### ISSUE TITLE
Generate C++ typesupport for actions

### ISSUE THREAD

**Issue by sloretz** [ros2/rosidl_typesupport_fastrtps#10]
#### Feature description

`rosidl_typesupport_fastrtps_c` needs to be aware of actions and generate typesupport for them.

##### CMake Code

The CMake code is expected to change with the IDL work, but at a minimum what's currently there needs to be able to recognize .action files.

https://github.com/ros2/rosidl_typesupport_fastrtps/blob/7263c0849a4c0d95276e19581d3a4814abcf3afb/rosidl_typesupport_fastrtps_cpp/cmake/rosidl_typesupport_fastrtps_cpp_generate_interfaces.cmake#L30-L31

##### Services in .action folder

The service templates would need to change to allow service definitions to live in the .actions folder

https://github.com/ros2/rosidl_typesupport_fastrtps/blob/7263c0849a4c0d95276e19581d3a4814abcf3afb/rosidl_typesupport_fastrtps_c/resource/srv__type_support_c.cpp.em#L27-L28

##### Parsing code

The parsing code needs a switch if the file is an action file, and if so call the appropriate parser for actions.

https://github.com/ros2/rosidl_typesupport_fastrtps/blob/7263c0849a4c0d95276e19581d3a4814abcf3afb/rosidl_typesupport_fastrtps_cpp/rosidl_typesupport_fastrtps_cpp/__init__.py#L33

connects to ros2/ros2#583
blocked by ros2/rosidl#305

**Comment by sloretz** [comment]
Closing in favor of just making rmw specific typesupports able to generate services in another folder

https://github.com/ros2/rosidl_typesupport_fastrtps/issues/9#issuecomment-431101549
